src/core/reaction_methods/ReactionAlgorithm.cpp

```diff
@@ -337,7 +337,8 @@ void ReactionAlgorithm::generic_oneway_reaction(
       current_reaction, E_pot_old, E_pot_new, old_particle_numbers);
 
   std::vector<double> exponential = {exp(-1.0 / kT * (E_pot_new - E_pot_old))};
-  current_reaction.accumulator_exponentials(exponential);
+  current_reaction.accumulator_potential_energy_difference_exponential(
+      exponential);
```

Comment:
I took the liberty of rebasing the PR to make the history cleaner. You can verify that your work was preserved in the new history with the command `git diff -w 4c28b517962ad267 df938c428b3`.

Here is the procedure I used to rebase the branch:
```sh
git checkout -b refactoring_reaction_ensamble_2_rebase fdea17683
git rebase -i HEAD
```
In the text editor, remove the word `noop` and write the following:
```
p 4d85439780c8e92
f 070406aff614e
f f88933efe5cb
f 9146d918ef5
f c2c82b90a
f d45895c5e072b59b8
f 4c28b517962a
p 2b5d07de3f
p 9952110b17
f 958033e424471a
f 26696ed13
```
Save the file and check that the history was preserved:
```sh
git diff fdea17683 refactoring_reaction_ensamble_2 # this shouldn't show any difference
```
After that, I split the `kT` renaming task from the deprecation of the `ConstantpHEnsemble` reaction coefficients task in two separate commits and applied the formatting script to every commit.